machine/v1/types_nutanixprovider.go

```diff
@@ -0,0 +1,244 @@
+package v1
+
+import (
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// NutanixCredentialsSecretName is the name of the secret holding the credentials for PC client
+	NutanixCredentialsSecretName = "nutanix-credentials"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NutanixMachineProviderConfig is the Schema for the nutanixmachineproviderconfigs API
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:openapi-gen=true
+type NutanixMachineProviderConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// clusterReferenceUuid is the UUID of the PE/cluster the Machine's VM will be created
+	ClusterReferenceUUID string `json:"clusterReferenceUuid,omitempty"`
+
+	// imageUuid is the UUID of the rhcos image uploaded to the PC.
+	// If the imageUUID is configured in the Machine CR, it will be used to create the VM.
+	// Otherwise, the imageName will be used to obtain the imageUUID, before creating the VM.
+	ImageUUID string `json:"imageUuid,omitempty"`
+
+	// imageName is the name of the rhcos image uploaded to the PC
+	ImageName string `json:"imageName,omitempty"`
+
+	// subnetUuid is the UUID of the network subnet to use for the Machine's VM
+	SubnetUUID string `json:"subnetUuid,omitempty"`
+
+	// numVcpusPerSocket is the number of vCPUs per socket of the VM to create
+	NumVcpusPerSocket int64 `json:"numVcpusPerSocket,omitempty"`
+
+	// numSockets is the number of sockets of the VM to create
+	NumSockets int64 `json:"numSockets,omitempty"`
+
+	// memorySizeMib is the memory size in megabytes of the VM to create
+	MemorySizeMib int64 `json:"memorySizeMib,omitempty"`
+
+	// diskSizeMib is the disk size in megabytes of the VM to create
+	DiskSizeMib int64 `json:"diskSizeMib,omitempty"`
+
+	// powerState is the expected power state of the VM to create
+	PowerState string `json:"powerState,omitempty"`
```

Comment:
Do we want users to be able to create VMs that are powered off? Does that create issues for OpenShift, eg, a Machine is created, but is never powered on. The Machine sits now in a `Provisioned` phase with no Node for a long time, this would trigger an alert to customers based on our current configuration of alerts

Reply:
Initially we have this field for cluster-api MachinePool use case. It seems we don't need other VM "powerState" than "ON" here with Openshift. I will remove this field.

Reply:
Removed.